### ISSUE TITLE
CycleGan-Turbo does not convert images

### ISSUE THREAD

**Issue by ClemensSchwarke** [GaParmar/img2img-turbo#53]
Hi,
when following the instructions for training CycleGan-Turbo with the horses&zebras dataset, the final network always outputs the reconstructed original image and is not able to translate between domains. Is this a behavior that often occurs when training?

Thanks in advance!

**Comment by onepeachbiubiubiu** [comment]
+1 , did you use bf16 or fp16?

**Comment by GaParmar** [comment]
Could you share the outputs and what dataset you are training on?

**Comment by onepeachbiubiubiu** [comment]
horse2zebra dataset, output images only reconstruct input image without translating to the other domain. I am using bf16 on RTX 4090, will it make any difference?

**Comment by tfriedel** [comment]
I observed the same issue. Tried with the proposed fix and without it. I think there's a bug in the code. Would really like to see the original code that hopefully doesn't include the bug.

**Comment by nldhuyen0047** [comment]
Hello, I'm facing a similar issue as you. I was wondering if you've found the problem and been able to fix it?

Thank you so much.

**Comment by tfriedel** [comment]
No I'm afraid not. I haven't continued with this. 
@GaParmar would be much appreciated if you could either fix the issue or share the source that was used to generate the results in the paper.

**Comment by GaParmar** [comment]
Could you try training without any mixed precision. 
I just tried running the example training I have provided in the readme today morning. 
I was able to match the performance of the results in the paper. 

This is the `accelerate config file` i use:
```
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: MULTI_GPU
downcast_bf16: 'no'
gpu_ids: 0,1,2,3
machine_rank: 0
main_training_function: main
mixed_precision: 'no'
num_machines: 1
num_processes: 4
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false
```

This is my exact training command:
```
export NCCL_P2P_DISABLE=1
accelerate launch --main_process_port 29501 --config_file temp/config_4gpu_full.sh \
    src/train_cyclegan_turbo.py \
    --pretrained_model_name_or_path="stabilityai/sd-turbo" \
    --output_dir="output/cyclegan_turbo/my_horse2zebra" \
    --dataset_folder "data/my_horse2zebra" \
    --train_img_prep "resize_286_randomcrop_256x256_hflip" --val_img_prep "no_resize" \
    --learning_rate="1e-5" --max_train_steps=25000 \
    --train_batch_size=1 --gradient_accumulation_steps=1 \
    --report_to "wandb" --tracker_project_name "gparmar_unpaired_h2z_cycle_debug_v3" \
    --enable_xformers_memory_efficient_attention --validation_steps 250 \
    --lambda_gan 0.5 --lambda_idt 1 --lambda_cycle 1
```



These are the results I see after training for ~ 15 mins. The horses do start getting converted to zebras. 
<img width="1105" alt="image" src="https://github.com/user-attachments/assets/8cb0f51a-499e-4dd3-a493-cf62c0eb1c04">

**Comment by GaParmar** [comment]
@tfriedel @nldhuyen0047 @onepeachbiubiubiu @ClemensSchwarke 
Could you share your exact `accelerate config` file and the training command you use for training? 
I can perhaps help debugging if I can recreate the training performance you are encountering :). 

Cheers, 
Gaurav

**Comment by tfriedel** [comment]
@GaParmar 
Thanks for looking into this!
I have tried your command and config and now it works. Difference was the mixed precision. Weird that it doesn't work with it on.

**Comment by GaParmar** [comment]
Yeah, I am slightly surprised by how big of a role mixed-precision training plays as well. 
I am open to any PR or suggestions about improving mixed-precision training.

**Comment by nldhuyen0047** [comment]
> @tfriedel @nldhuyen0047 @onepeachbiubiubiu @ClemensSchwarke Could you share your exact `accelerate config` file and the training command you use for training? I can perhaps help debugging if I can recreate the training performance you are encountering :).
> 
> Cheers, Gaurav

This is my `command line` using for training:
export NCCL_P2P_DISABLE=1
accelerate launch --main_process_port 29501 src/train_cyclegan_turbo.py \
    --pretrained_model_name_or_path="stabilityai/sd-turbo" \
    --output_dir="outputs/test" \
    --dataset_folder "data/test" \
    --train_img_prep "resize_286_randomcrop_256x256_hflip" --val_img_prep "no_resize" \
    --learning_rate="1e-5" --max_train_steps=25000 \
    --train_batch_size=1 --gradient_accumulation_steps=1 \
    --report_to "wandb" --tracker_project_name "gparmar_unpaired_h2z_cycle_debug_v2" \
    --mixed_precision "bf16" \
    --enable_xformers_memory_efficient_attention --validation_steps 250 \
    --lambda_gan 0.5 --lambda_idt 1 --lambda_cycle 1

Thank you so much.

**Comment by GaParmar** [comment]
Hi @nldhuyen0047, 

Could you try removing `--mixed_precision "bf16"` from the training command? 

-Gaurav

**Comment by nldhuyen0047** [comment]
I tried removing --mixed_precision "bf16", and the code ran, but, I have encountered another problem, my training code stopped after several steps (800/25000).
